hibernate-core/src/main/java/org/hibernate/engine/spi/CollectionEntry.java

```diff
@@ -59,89 +63,109 @@ public final class CollectionEntry implements Serializable {
 	private transient CollectionPersister currentPersister;
 	private transient Object currentKey;
```

Comment:
Shouldn't that be nullable as well?

```suggestion
	private transient @Nullable CollectionPersister currentPersister;
	private transient @Nullable Object currentKey;
```

Reply:
Indeed.